src/lib/core/redux/slices/rtcAnalytics.ts

```diff
@@ -0,0 +1,202 @@
+import { PayloadAction, createSlice } from "@reduxjs/toolkit";
+import { RootState } from "../store";
+import { createAppThunk } from "../thunk";
+import { createReactor, startAppListening } from "../listenerMiddleware";
+import { selectRtcConnectionRaw, selectRtcManagerInitialized, selectRtcStatus } from "./rtcConnection";
+import { selectAppDisplayName, selectAppExternalId } from "./app";
+import { selectOrganizationId } from "./organization";
+import { selectLocalParticipantUserRole, selectSelfId } from "./localParticipant";
+import { selectSignalStatus } from "./signalConnection";
+import { selectDeviceId } from "./deviceCredentials";
+import {
+    selectIsCameraEnabled,
+    selectIsMicrophoneEnabled,
+    selectLocalMediaStream,
+    selectScreenshareStream,
+} from "./localMedia";
+
+type RtcAnalyticsCustomEvent = {
+    actionType: string;
+    rtcEventName: string;
+    getValue: (state: RootState) => unknown;
+    getOutput: (value: unknown) => unknown;
+};
+
+export const rtcAnalyticsCustomEvents: { [key: string]: RtcAnalyticsCustomEvent } = {
```

Comment:
I assume all of these are just copied over from the PWA more or less

Reply:
Yep names and return structure are, the actions & selectors needed to be changed accordingly 👍